src/spyglass/utils/mixins/export.py

```diff
@@ -0,0 +1,333 @@
+            "_load_admin",  # Prevents on permission check
+        ]  # if called by any in banned, return
+        if set(banned) & self._called_funcs():
+            return
+
+        restr = restriction or self.restriction or True
+        limit = kwargs.get("limit")
+        offset = kwargs.get("offset")
+        if limit or offset:  # Use result as restr if limit/offset
+            restr = self.restrict(restr).fetch(
+                log_export=False, as_dict=True, limit=limit, offset=offset
+            )
+
+        restr_str = make_condition(self, restr, set())
+
+        if restr_str is True:
+            restr_str = "True"  # otherwise stored in table as '1'
+
+        if isinstance(restr_str, str) and "SELECT" in restr_str:
+            raise RuntimeError(
+                "Export cannot handle subquery restrictions. Please submit a "
+                + "bug report on GitHub with the code you ran and this"
+                + f"restriction:\n\t{restr_str}"
+            )
+
+        if isinstance(restr_str, str) and len(restr_str) > 2048:
+            raise RuntimeError(
+                "Export cannot handle restrictions > 2048.\n\t"
+                + "If required, please open an issue on GitHub.\n\t"
+                + f"Restriction: {restr_str}"
+            )
+
+        if isinstance(restr_str, str):
+            restr_str = bash_escape_sql(restr_str, add_newline=False)
+
+        if restr_str in self._export_cache[self.full_table_name]:
+            return
+        self._export_cache[self.full_table_name].add(restr_str)
+
+        self._export_table.Table.insert1(
+            dict(
+                export_id=self.export_id,
+                table_name=self.full_table_name,
+                restriction=restr_str,
+            )
+        )
+        restr_logline = restr_str.replace("AND", "\n\tAND").replace(
+            "OR", "\n\tOR"
+        )
+        logger.debug(f"\nTable: {self.full_table_name}\nRestr: {restr_logline}")
+
+    def _log_fetch_nwb(self, table, table_attr):
+        """Log fetch_nwb for export table."""
+        tbl_pk = "analysis_file_name"
+        fnames = self.fetch(tbl_pk, log_export=True)
+        logger.debug(
+            f"Export: fetch_nwb\nTable:{self.full_table_name},\nFiles: {fnames}"
+        )
+        self._export_table.File.insert(
+            [{"export_id": self.export_id, tbl_pk: fname} for fname in fnames],
+            skip_duplicates=True,
+        )
+        fnames_str = "('" + "', ".join(fnames) + "')"  # log AnalysisFile table
+        table()._log_fetch(restriction=f"{tbl_pk} in {fnames_str}")
+
+    def _run_join(self, **kwargs):
+        """Log join for export.
+
+        Special case to log primary keys of each table in join, avoiding
+        long restriction strings.
+        """
+        table_list = [self]
+        other = kwargs.get("other")
+
+        if hasattr(other, "_log_fetch"):  # Check if other has mixin
+            table_list.append(other)  # can other._log_fetch
```

Comment:
Will hit an error here if user runs without making an instance of other
ex)
```python
Session() * IntervalList() #runs ok
Session() * IntervalList #hits error
```
The error is triggered because the log_fetch call below doesn't have a `self` argument if `other` wasn't instantiated.

To fix
```suggestion
            if isinstance(other, TableMeta):
                table_list.append(other())  # can other._log_fetch
            else:
                table_list.append(other)  # can other._log_fetch
```

and add `from datajoint.user_tables import TableMeta` at the top of the file

Reply:
Thanks @samuelbray32, good catch. I took a slightly different approach following something in the dj source code that should serve the same function of initializing `other`